### PR TITLE
refactor!: remove --config-path from CLI

### PR DESCRIPTION
**Pull request by tromai**
Closes https://github.com/oracle/macaron/issues/417

### CODE REVIEW

File: src/macaron/dependency_analyzer/cyclonedx.py [hunk 1]
```diff
@@ -259,46 +258,26 @@ def add_latest_version(
...
-        config_deps : list[Configuration]
-            Dependencies defined in the configuration file.
         resolved_deps : dict[str, DependencyInfo]
             The automatically resolved dependencies.
 
         Returns
         -------
         list[Configuration]
-            The result list contains the merged dependencies.
+            The dependencies list to be used by the Analyzer.
```
Comment (benmss) [inline src/macaron/dependency_analyzer/cyclonedx.py:272]:
```suggestion
            The dependency list to be used by the Analyzer.
```

File: tests/integration/cases/micronaut-projects_micronaut-test/test.yaml [hunk 2]
```diff
@@ -2,25 +2,40 @@
 # Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
 
 description: |
-  Test configurations for micronaut-projects/micronaut-test.
+  Test micronaut-projects/micronaut-test. qos-ch/slf4j and ben-manes/caffeine were originally used as manually specified dependencies.
+  We still keep it here even though manual dependencies specification is not available anymore.
```
Comment (benmss) [inline tests/integration/cases/micronaut-projects_micronaut-test/test.yaml:6]:
```suggestion
  We still keep it here even though manual dependency specification is not available anymore.
```

Comment (tromai) [reply tests/integration/cases/micronaut-projects_micronaut-test/test.yaml]: Fixed in https://github.com/oracle/macaron/pull/844/commits/d00d4827d3f0f998788f91d9eadbe24a488464ca

File: src/macaron/dependency_analyzer/cyclonedx.py (hunk 1 above)
Comment (tromai) [reply src/macaron/dependency_analyzer/cyclonedx.py]: Fixed in https://github.com/oracle/macaron/pull/844/commits/d00d4827d3f0f998788f91d9eadbe24a488464ca

**Review by benmss — APPROVED** [review]
Looks good to me.

**Review by behnazh-w — APPROVED** [review]

**Pull request merged**